components/common/src/command/package/install.rs

```diff
@@ -732,10 +732,28 @@ impl<'a> InstallTask<'a> {
                 let temp_dir = temp_package_directory(real_install_path)?;
                 let temp_install_path = &pkg_install_path(ident, Some(temp_dir.path()));
                 artifact.unpack(Some(temp_dir.path()))?;
-                fs::rename(temp_install_path, real_install_path)?;
+
+                match fs::rename(temp_install_path, real_install_path) {
```

Comment:
Not a big deal, but you could avoid the skipped `Ok(…)` with an `if let` like:

```suggestion
                if let Err(e) = fs::rename(temp_install_path, real_install_path) {
                    // ...
                }
```

Reply:
I think you can replace this whole construct with:
```rust
                fs::rename(temp_install_path, real_install_path).or_else(|e| {
                    debug!("rename failed with {:?}, checking for installed package", e);
                    PackageInstall::load(ident.as_ref(), Some(self.fs_root_path)).map(|_| ())
                })?;
```

Reply:
How do you feel about the `if let` change I've pushed vs the or_else and map.  

To be honest, I don't think `.map(|_| ())})?;` adds much to the clarity of this code.  For example, the current code is returning the error from the rename if the package lookup fails. Does this example still do that?  To figure it out I have to pretty carefully parse rather dense code.

I'm happy to change it if y'all feel strongly about it though.

Reply:
I think the `if let` is an improvement, but I find it still less idiomatic than the rename/or_else/load change that I suggested. I agree that it's a bit dense, but that may be a matter of rust familiarity. Conceptually, it's a perfect match for what we're trying to say here: try a `rename`, `or else` if that fails, do a `load`. I think the tradeoff with density is that this more FP-oriented approach makes a large class of logic errors less likely.

You're right that the `.map(|_| ())` doesn't add anything; it's just required to discard the information that `load` returns in the success case. It's rather ugly, and we could avoid it if we added an appropriate `From` implementation to convert the `Result` type. How much better does this look to you?

```rust
fs::rename(temp_install_path, real_install_path).or_else(|e| {
    debug!("rename failed with {:?}, checking for installed package", e);
    PackageInstall::load(ident.as_ref(), Some(self.fs_root_path))
})?;
```

All that said, if you prefer the version you have now, I'm happy to go with it; I don't want to be imperious about style.